lib/utils.pm

```diff
@@ -1362,6 +1362,11 @@ sub reconnect_mgmt_console {
         }
 
         if (!check_var('DESKTOP', 'textmode')) {
+            if (check_var("UPGRADE", "1") && is_sle('15+') && is_sle('<15', get_var('HDDVERSION'))) {
+                select_console 'root-console';
+                record_soft_failure('bsc#1154156 - After upgrade from 12SP5, SuSEfirewall2 blocks xvnc.socket on s390x');
+                systemctl('stop SuSEfirewall2');
+            }
```

Comment:
Again, this is not checking the error condition so the soft failure can be recorded regardless of the bug being present or not. As you are stopping SuSEfirewall2 as mitigation, probably you can check if the corresponding VNC ports are open or blocked. If they are blocked then this should qualify as the bug being detected and then you can unblock the port (not stop the firewall).